src/components/ScriptExplorer/OutputsForm.jsx

```diff
@@ -105,22 +103,28 @@ class OutputsForm extends React.Component {
   };
 
   outputsAndFeeTotal = () => {
-    const { outputs, fee, inputs, updatesComplete } = this.props;
-    if (!inputs.length) return "";
-
-    const total = outputs
-      .map((output) => new BigNumber(output.amount || 0))
+    const { outputs, fee, updatesComplete, inputs } = this.props;
+    let total = outputs
+      .map((output) => {
+        let { amount } = output;
+        // eslint-disable-next-line no-restricted-globals
+        if (!amount || !amount.length || isNaN(amount)) amount = 0;
+        return new BigNumber(amount);
+      })
       .reduce(
         (accumulator, currentValue) => accumulator.plus(currentValue),
         new BigNumber(0)
-      )
-      .plus(new BigNumber(fee));
+      );
+
+    // only care to add fee if we have inputs
+    // which we won't have in auto-spend wallet output form
+    if (fee && inputs.length) total = total.plus(fee);
 
     if (updatesComplete) {
       this.outputsTotal = total;
-      return total;
+      return total.toFixed(8);
     }
-    return this.outputsTotal;
+    return "0.00000000";
```

Comment:
Why is this a string?  It looks like in other cases we return `new BigNumber(0)`?

Reply:
toFixed returns a string, so this is the fallback. This function is for returning a "displayable" version of the number. Before the toFixed was being done in the render function (and added in multiple places), since it seemed like this method was only being used for determining a value for the view, this seemed like the cleanest approach.